crates/syntax/src/ast/edit_in_place.rs

```diff
@@ -164,3 +164,29 @@ impl ast::Use {
         ted::remove(self.syntax())
     }
 }
+
+impl ast::Impl {
+    /// Add a node to at the start of the impl block.
+    pub fn push_front(&self, item: ast::AssocItem) {
+        let l_curly = self.get_or_create_assoc_item_list().l_curly_token().unwrap();
```

Comment:
I believe we should also have a `get_or_create` variant for the curlies here. I don't think the parser can create an assoc list without them but we shouldn't rely on that I imagine?

Reply:
Maybe `get_or_create_assoc_item_list` should create the curlies in that case?

Reply:
Yeah, the current code reads OK to me. I am not sure that's what we'll end up using long term, but I don't think we have enough data to decide one way or the other.